sia/core.go

```diff
@@ -138,14 +138,16 @@ func CreateCore(config Config) (c *Core, err error) {
 	}
 	*/
 
-	// Bootstrap to the network.
-	err = c.initializeNetwork(config.ServerAddr, config.Nobootstrap)
-	if err == network.ErrNoPeers {
-		// log.Println("Warning: no peers responded to bootstrap request. Add peers manually to enable bootstrapping.")
-	} else if err != nil {
-		return
-	}
-	// c.host.Settings.IPAddress = c.server.Address()
+	// Bootstrap to the network (may take a few seconds).
+	go func() {
+		err = c.initializeNetwork(config.ServerAddr, config.Nobootstrap)
+		if err == network.ErrNoPeers {
+			// log.Println("Warning: no peers responded to bootstrap request. Add peers manually to enable bootstrapping.")
+		} else if err != nil {
+			return
+		}
+		// c.host.Settings.IPAddress = c.server.Address()
+	}()
```

Comment:
We talked about it, but just to reiterate:

the go func() should probably happen inside of the initializeNetwork(), because some things potentially need to be initialized before you continue processing stuff. You don't need to find peers right away.